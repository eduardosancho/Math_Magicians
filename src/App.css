.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 10vh;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.calculator-screen { grid-area: result; }
.divide { grid-area: divideBtn; }
.multiply { grid-area: multiplyBtn; }
.substract { grid-area: substractBtn; }
.add { grid-area: addBtn; }
.equal-sign { grid-area: equalBtn; }
.all-clear { grid-area: clearBtn; }
.invert-sign { grid-area: invertBtn; }
.percentage { grid-area: percentageBtn; }
.zero { grid-area: zeroBtn; }
.one { grid-area: oneBtn; }
.two { grid-area: twoBtn; }
.three { grid-area: threeBtn; }
.four { grid-area: fourBtn; }
.five { grid-area: fiveBtn; }
.six { grid-area: sixBtn; }
.seven { grid-area: sevenBtn; }
.eight { grid-area: eightBtn; }
.nine { grid-area: nineBtn; }
.decimal { grid-area: decimalBtn; }

.calculator {
  margin: 0 10vw;
  border: solid black 1px;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(6, 1fr);
  gap: 0;
  grid-auto-flow: row;
  grid-template-areas:
    "result result result result"
    "clearBtn invertBtn percentageBtn divideBtn"
    "sevenBtn eightBtn nineBtn multiplyBtn"
    "fourBtn fiveBtn sixBtn substractBtn"
    "oneBtn twoBtn threeBtn addBtn"
    "zeroBtn zeroBtn decimalBtn equalBtn";
}

.calculator > * {
  border-radius: 0%;
  border: solid rgba(155, 155, 155, 0.2) 0.5px;
}

#result-display {
  grid-column: 1 / 5;
  grid-row: 1;
  text-align: end;
  padding-right: 10px;
  background-color: gray;
}
