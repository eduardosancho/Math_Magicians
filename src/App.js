import './App.css';
import Calculator from './components/Calculator';

function App() {
  return (
    <div className="App">
      <header className="App-header">Header</header>
      <Calculator />
    </div>
  );
}

export default App;
