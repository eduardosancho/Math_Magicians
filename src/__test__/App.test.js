import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';

describe('Test Router behaviour', () => {
  it('When user presses Quote it will render Quote page', () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    const quoteLink = screen.getByText('Quote');
    fireEvent.click(quoteLink);
    const main = container.querySelector('main');
    expect(main).toMatchSnapshot();
  });
});
