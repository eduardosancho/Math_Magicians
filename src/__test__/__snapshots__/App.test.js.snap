// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Router behaviour When user presses Quote it will render Quote page 1`] = `
<main>
  <div
    class="quote-container"
  >
    <p>
      Mathematics is not about numbers, equations, computations, or algorithms; it is about understanding. -William Paul Thurston
    </p>
  </div>
</main>
`;
